app/controllers/forums/forums_controller.rb

```diff
@@ -142,7 +142,7 @@ def toggle_lock
 
   private
   def load_forum
-    @forum = ForumForum.find_using_slug(params[:id] || params[:forum_id])
+    @forum = @course.forums.find_using_slug(params[:id] || params[:forum_id])
```

Comment:
We have <code> raise ActiveRecord::RecordNotFound unless @forum </code> already.